mcp/src/main/java/io/airlift/mcp/reference/ReferenceFilter.java

```diff
@@ -10,31 +17,61 @@
 import jakarta.servlet.http.HttpServletResponse;
 
 import java.io.IOException;
+import java.util.Optional;
 
+import static com.google.common.net.HttpHeaders.WWW_AUTHENTICATE;
+import static io.modelcontextprotocol.server.transport.HttpServletStatelessServerTransport.APPLICATION_JSON;
 import static java.util.Objects.requireNonNull;
 
 public class ReferenceFilter
         extends HttpFilter
 {
+    public static final String MCP_IDENTITY_ATTRIBUTE = "airlift.mcp.identity";
```

Comment:
Please make private and add a public static method that extracts the value from a request.